File: backend/app/schemas.py
from pydantic import BaseModel
from typing import List, Optional, Literal

class ValidationError(BaseModel):
    row_index: Optional[int] = None
    id: Optional[int] = None
    column: Optional[str] = None
    error_message: str

class ValidationResponse(BaseModel):
    status: Literal["pass", "fail"]
    errors: List[ValidationError]

File: rust-backend/src/schemas.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ValidationError {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub row_index: Option<usize>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub column: Option<String>,
    pub error_message: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ValidationResponse {
    pub status: String,
    pub errors: Vec<ValidationError>,
}
